packages/components/src/base-control/index.js

```diff
@@ -14,6 +14,26 @@ import {
 	StyledHelp,
 } from './styles/base-control-styles';
 
+/**
+ * @typedef Props
+ * @property {string} id The id of the element to which labels and help text are being generated.
+ *                       That element should be passed as a child.
+ * @property {import('react').ReactNode} help If this property is added, a help text will be
```

Comment:
Out of curiosity (rather than thinking it _should_ be done) why don't we alias this type like we do the rest of them in the `@wordpress/element` package?

Reply:
You're referring to how we alias e.g. `ReactElement` as `WPElement`? I don't see any value in that and would prefer to use the types we're referring to more directly.

Additionally, a lot of the aliases lose the type parameters, which is unfortunate.

That's the only _why_ for this and is very much tied to the question of whether or not we _should_ 🙃

Reply:
> Additionally, a lot of the aliases lose the type parameters, which is unfortunate.

I noticed too that when I tried to alias `ReactNode` it lost some of the type itself for some reason (it was no longer able to be `null` for example). Do you know why that is?

Reply:
I don't understand why that one would be affected. The problem with type parameters is that we don't provide them in the aliases at the moment. We'd need to rewrite them as

```diff
diff --git a/packages/element/src/react.js b/packages/element/src/react.js
index 520401dcc5..2bb42c77b4 100644
--- a/packages/element/src/react.js
+++ b/packages/element/src/react.js
@@ -37,7 +37,8 @@ import { isString } from 'lodash';
 /**
  * Object containing a React component.
  *
- * @typedef {import('react').ComponentType} WPComponent
+ * @template P
+ * @typedef {import('react').ComponentType<P>} WPComponent
  */
 
 /**
```

This _also_ isn't great because now we have a required type parameter. I don't think there's a good way to write type parameter (@template) defaults in JSDoc at this time.